src/internal/connector/sharepoint/restore.go

```diff
@@ -0,0 +1,62 @@
+package sharepoint
+
+import (
+	"context"
+
+	"github.com/alcionai/corso/src/internal/connector/graph"
+	"github.com/alcionai/corso/src/internal/connector/onedrive"
+	"github.com/alcionai/corso/src/internal/connector/support"
+	"github.com/alcionai/corso/src/internal/data"
+	"github.com/alcionai/corso/src/pkg/backup/details"
+	"github.com/alcionai/corso/src/pkg/control"
+	"github.com/alcionai/corso/src/pkg/path"
+	"github.com/pkg/errors"
+)
+
+// RestoreCollections will restore the specified data collections into OneDrive
+func RestoreCollections(
+	ctx context.Context,
+	service graph.Service,
+	dest control.RestoreDestination,
+	dcs []data.Collection,
+	deets *details.Details,
```

Comment:
SharePoint doesn't have a way to restore via the GraphAPI. Would need to verify that this library would work via the current APIs for a sharepoint drive.  This might require a mock collection to be added prior to this PR.

Reply:
GraphAPI doesn't have write control for _sites_, but it does interface with _driveItems_.  Since all libraries and library items are handled through the drives api, there's no blocker on using the graph api (or, more accurately, our graph-constructed api calls) for restoration.